common/primitives/core/src/intent.rs

```diff
@@ -47,3 +51,27 @@ pub struct TransferSolana {
     pub to: [u8; 32],
     pub value: u64,
 }
+
+#[derive(Encode, Decode, Debug, Clone, PartialEq, Eq, MaxEncodedLen, TypeInfo)]
+pub struct Asset {
+    pub id: AssetId,
```

Comment:
This is u128 - does it mean we will have an asset registry ?
Maybe we need erc20 token addreses ?

Reply:
Good question, I intentionally set it like this to open a discussion. We may need to have some sort of registry we can use to map the assets with the CEX (e.g binance convert api)

Reply:
I can imagine we have a registry which maps (network, CA) -> token id on binance somewhere in binance module